.github/workflows/trigger-publish.yml

```diff
@@ -39,4 +39,30 @@ jobs:
           REPO: ${{ toJSON('temporal') }}
           BRANCH: ${{ toJSON(steps.get_branch.outputs.branch) }}
         run: |
+          # Dispatch the workflow
           curl -fL -X POST -H "Accept: application/vnd.github.v3+json" -H "Authorization: token $PAT" "https://api.github.com/repos/$PARENT_REPO/actions/workflows/$WORKFLOW_ID/dispatches" -d '{"ref":'"$PARENT_BRANCH"', "inputs": { "repo":'"$REPO"', "branch":'"$BRANCH"' }}'
+          
+          # Wait a few seconds for the workflow to start
+          sleep 10
```

Comment:
This seems error-prone. And the "get latest run" is racy. Could we do something like:
- get the latest run
- dispatch the workflow
- in a loop:
  - get the last bunch of runs, filter out anything <= the original latest run
  - if we find one, break
  - if we find more than one... can we examine its properties to see if it's the same one we triggered?
- then poll